### PR TITLE
koji_promote: remove obsolete koji_promote plugin

### PR DESCRIPTION
**Pull request by mlangsdorf**
Fixes OSBS-3863

The koji_promote plugin is now obsolete, so remove it entirely.

Signed-off-by: Mark Langsdorf <mlangsdo@redhat.com>

Maintainers will complete the following section:
- [x] Commit messages are descriptive enough
- [x] "Signed-off-by:" line is present in each commit
- [x] Code coverage from testing does not decrease and new code is covered
- [x] JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- [x] Pull request includes link to an osbs-docs PR for user documentation updates
- [x] New feature can be disabled from a configuration file

### CODE REVIEW

File: atomic_reactor/plugins/exit_koji_tag_build.py [hunk 1]
```diff
@@ -77,11 +76,8 @@ def run(self):
 
         build_id = self.workflow.exit_results.get(KojiImportPlugin.key)
         if not build_id:
-            build_id = self.workflow.exit_results.get(KojiPromotePlugin.key)
-            if not build_id:
-                self.log.info('No koji build from %s or %s', KojiImportPlugin.key,
-                              KojiPromotePlugin.key)
-                return
+            self.log.info('No koji build from %s', KojiImportPlugin.key)
+            return
```
Comment (twaugh) [inline atomic_reactor/plugins/exit_koji_tag_build.py]: Looks like this block is not covered by unit testing.

Comment (mlangsdorf) [reply atomic_reactor/plugins/exit_koji_tag_build.py]: Fixed.

**Comment by mlangsdorf** [comment]
Moved the KojiUploadLogger tests into test_koji_utils where they belong.

**Comment by mlangsdorf** [comment]
Not adding to release notes as this is the removal of a code path that hasn't be accessible in a while anyway.

**Pull request merged**